static/js/main.js

```diff
@@ -0,0 +1,698 @@
+
+  /*
+   * Register blur listener to re-generate/re-draw D3 graph when a
+   * step name input field looses focus
+   */
+  $(document).on("blur", ".ssc-steps .ssc-step input[name='step_name[]']",
+    function(evt){
+      draw_graph(generate_graph_from_ssc_steps());
+  });
+
+  /*
+   * Register custom removed event listener to re-generate/re-draw D3 graph
+   * when a step form element got removed
+   */
+  $(document).on("custom.removed",
+    function(evt, removed_elem_class){
+      if (removed_elem_class == ".ssc-step")
+        draw_graph(generate_graph_from_ssc_steps());
+  });
+
+  /*
+   * Register sortable sortupdate event listener to re-generate/re-draw D3
+   * graph when the step form elements got re-ordered
+   */
+  $(".sort-container.ssc-steps").on("sortupdate",
+    function(evt) {
+      draw_graph(generate_graph_from_ssc_steps());
+  });
+
+  /*
+   * Initialize sortable plugin on `.sort-container`s to enable
+   * drag and drop sorting.
+   * Note:
+   * Needs to be re-initialized when elements are added to the DOM
+   */
+   sortable(".sort-container", {
+      forcePlaceholderSize: true,
+      placeholderClass: "sort-placeholder",
+   });
+
+  /*
+   * Register click listener that turns a pre html element into a textarea
+   * and immediately selects the contents
+   * This is a neat little feature for copy-paste snippets
+   */
+  $("pre.code").click(function(){
+    // Extract (copy) all attributes of the clicked pre element
+    var attrs = {};
+    $.each(this.attributes, function(idx, attr) {
+        attrs[attr.nodeName] = attr.nodeValue;
+    });
+
+    // We don't need red squiggly underlines for code snippets
+    attrs["spellcheck"] = "false";
+
+    // Create new empty element with above copied attributes, set it to
+    // readonly and insert the text content from the clicked pre element
+    $textarea = $("<textarea />", attrs)
+      .height($(this).outerHeight())
+      .prop("readonly", true)
+      .append($(this).html());
+
+    // Remove old element from DOM and insert new
+    $(this).replaceWith($textarea);
+
+    // Select the contents
+    $textarea.select();
+  });
+});
+
+/*****************************************************************
+ * Miscellaneous functions globally available
+ ****************************************************************/
+
+/*
+ * Retrieve the CSRF token written by the server to a meta element in the page
+ * header and format it so that it can be used as additional header for form
+ * posting ajax calls
+ */
+function _get_csrf_token_header() {
+  return {
+    'X-CSRFToken': $('meta[name="csrftoken"]').attr('content')
+  };
+}
+
+/*
+ * Append passed message to DOM (using message template), and show for
+ * a fixed amount of type (5 seconds) the second parameter is used as style
+ * class.
+ *
+ * See https://v4-alpha.getbootstrap.com/components/alerts/ for available
+ * types and styles.
+ *
+ * TODO:
+ * Sometimes 5 seconds are to much and sometimes not enough
```

Comment:
Is this a "sometimes the messages take longer than they should and sometimes they take less" in a sense that the code doesn't stay for 5 secs? or is it more of a "it should figure out the timeout with some context or a variable that's passed to it"?